### PR TITLE
Fix #1836: BigInteger.ONE.gcd(x) loops forever

### LINKED ISSUE #1836: BigInteger.ONE.gcd(x) loops forever
A call to `BigInteger.ONE.gcd(x)` never returns. 

I have a PR ready for this, creating issue to reference against first

### PR DESCRIPTION
**Pull request by InTheNow**

### CODE REVIEW

**Comment by gzm0** [comment]
Jenkins, ok to test

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/math/BigIntegerModPowTest.scala [hunk 1]
```diff
@@ -15,6 +15,24 @@ object BigIntegerModPowTest extends JasmineTest {
 
   describe("BigIntegerModPowTest") {
 
+    it("testsGcdFirstOne") {
+      val aBytes = Array[Byte](1, 0, 0, 0, 0)
+      val big = new  BigInteger(1, aBytes)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/math/BigIntegerModPowTest.scala]: Double space.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/math/BigIntegerModPowTest.scala [hunk 2]
```diff
@@ -15,6 +15,24 @@ object BigIntegerModPowTest extends JasmineTest {
...
+      expect(BigInteger.ONE.gcd(big).intValue).toEqual(1)
+      expect(BigInteger.ONE.gcd(BigInteger.ZERO).intValue).toEqual(1)
+    }
+
+    it("testsGcdSecondOne") {
+      val aBytes = Array[Byte](1, 0, 0, 0, 0)
+      val big = new  BigInteger(1, aBytes)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/math/BigIntegerModPowTest.scala]: Same

File: javalib/src/main/scala/java/math/BigInteger.scala [hunk 3]
```diff
@@ -456,7 +456,8 @@ class BigInteger extends Number with Comparable[BigInteger] {
       val2
     } else if (val2.signum() == 0) {
       val1
-    } else if ((val1.numberLength == 1) && (val1.digits(0) > 0)) {
+    } else if (((val1.numberLength == 1) && (val1.digits(0) > 0))
+          && ((val2.numberLength == 1) && (val2.digits(0) > 0))) {
```
Comment (sjrd) [inline javalib/src/main/scala/java/math/BigInteger.scala]: `&&` should be at the end of the previous line. And this line should be indented 4 spaces wrt `}`.

**Comment by sjrd** [comment]
That's all.

**Comment by InTheNow** [comment]
Done

**Comment by sjrd** [comment]
LGTM

**Pull request merged**